src/AutoRest.CSharp/Common/Input/InputTypes/InputModelType.cs

```diff
@@ -4,11 +4,12 @@
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Linq;
 using System.Runtime.CompilerServices;
 
 namespace AutoRest.CSharp.Common.Input
 {
-    internal record InputModelType(string Name, string? Namespace, string? Accessibility, string? Deprecated, string? Description, InputModelTypeUsage Usage, IReadOnlyList<InputModelProperty> Properties, InputModelType? BaseModel, IReadOnlyList<InputModelType> DerivedModels, string? DiscriminatorValue, string? DiscriminatorPropertyName, InputDictionaryType? InheritedDictionaryType, bool IsNullable)
+    internal record InputModelType(string Name, string? Namespace, string? Accessibility, string? Deprecated, string? Description, InputModelTypeUsage Usage, IReadOnlyList<InputModelProperty> Properties, InputModelType? BaseModel, IReadOnlyList<InputModelType> DerivedModels, string? DiscriminatorValue, string? DiscriminatorPropertyName, InputDictionaryType? InheritedDictionaryType, bool IsNullable, IReadOnlyList<InputModelType>? Parents = null, bool IsBasePolyType = false)
```

Comment:
`Parents` reads quite confusing, could we use the `CompositionModels` as Alexander introduced previously to represent multiple `allOf` in swaggers?

Reply:
`CompositionModels` only represents `Immediate` parents, not `All` parents. If you think the naming is better, I can use that instead and change the value accordingly.

Reply:
Renamed `Parents` to `AllBaseModels`, and get `CompositionModels` from `AllBaseModels`

Reply:
Removed all of these, build properties in advance in `CodeModelConverter`